pkg/endpoint/endpoint.go

```diff
@@ -1475,6 +1477,45 @@ func (e *Endpoint) getIDandLabels() string {
 	return fmt.Sprintf("%d (%s)", e.ID, labels)
 }
 
+// MetadataResolverCB provides an implementation for resolving the endpoint
+// metadata for an endpoint such as the associated labels and annotations.
+type MetadataResolverCB func(*Endpoint) (identityLabels labels.Labels, infoLabels labels.Labels, annotations map[string]string, err error)
```

Comment:
I'd like to make a metadata structure for these return results at some point, but I figured that can always be handled in a non-bugfix PR in future.